### DISCUSSION TITLE
Row level security?

### DISCUSSION THREAD

**Discussion by man-tax** [Budibase/budibase#12947] [Help] [answered]
Hi team,

checking in after a while to see if there is any progress on row level security in Budibase? #4789 

Thanks
man-tax

**Comment by ConorWebb96** [comment]
Hey @man-tax,

I suppose you could achieve this functionality with the new [user](https://docs.budibase.com/docs/users-1) column. Basically you'd assign users per row with this and only query columns relevant to the logging in user on the published/design end. This would allow you to only show rows that are related to the user.

Let me know if this solves your issue.

**Reply by man-tax** [reply]
Thanks for the reply. This isn't restricting the backend query, is it? My issue is exactly as described in #4789 

> I would like my clients' users to be able to see their own record and all of their own invoices without giving them access to the entire invoices/clients table.

**Reply by ConorWebb96** [reply, marked as answer]
Hey @man-tax,

The only people whom would have access to all apps/records would be account admins. For basic users you can definitely add the level of restriction talked about in #4789. You would just need to setup your data providers/forms other components in such a way that it only allows the current user to see their own records related to the [user](https://docs.budibase.com/docs/users-1) column attached to each table row.

That way you could display only the customer details, invoices and other information related to who ever the logged in user is.

Hopefully this answers your question.

I'm going to close that issue out as I believe it has been resolved due to the [user](https://docs.budibase.com/docs/users-1) column.
